### PR TITLE
Implemented home, stop, get_status, get_data commands

### PR DESCRIPTION
**Pull request by noaht11**
- Setup proper state tracking in the Arduino firmware
 - Implemented the commands on both the MessageTerminal and firmware
 side
 - Added reset command to MessageTerminal, to close then re-open the
 serial connection
 - Added extra flush to MessageTerminal right after opening the serial
 port to avoid capturing any stray messages

### CODE REVIEW

File: firmware/lib/Gantry/Axis.cpp [hunk 1]
```diff
@@ -111,18 +129,22 @@ void isr_limit_switch_x()
...
-            axis_x.homing = 0;
-            DEBUG_PRINT_VAL("X axis has been homed - SUCCESS", 1);
+        if (axis_x.homing) {
+            if (status == RELEASED) {
+                axis_x.encoder.current = 0;
+                axis_x.homing = false;
+                DEBUG_PRINT_VAL("X axis has been homed - SUCCESS", 1);
+            }
+            else if (status == PRESSED) {
+                DEBUG_PRINT_VAL("HOMING IN POSITIVE DIR ", 0);
```
Comment (chuandudx) [inline firmware/lib/Gantry/Axis.cpp:142]: What does the `1` and `0` mean in `DEBUG_PRINT_VAL`?

File: firmware/lib/Gantry/Axis.cpp [hunk 2]
```diff
@@ -111,18 +129,22 @@ void isr_limit_switch_x()
...
+                home_axis(&axis_x);
+            }
         }
-        else if (axis_x.homing && (status == PRESSED)) {
-            DEBUG_PRINT_VAL("HOMING IN POSITIVE DIR ", 0);
-            home_axis(&axis_x);
+        else {
+            axis_x.fault = true;
```
Comment (chuandudx) [inline firmware/lib/Gantry/Axis.cpp:147]: good call on the fault implementation!

File: firmware/lib/Gantry/Axis.cpp [hunk 3]
```diff
@@ -204,18 +226,22 @@ void isr_limit_switch_y()
     uint32_t interrupt_time = millis();
     // If interrupts come faster than 10ms, assume it's a bounce and ignore
     if (interrupt_time - last_interrupt_time > 50) {
```
Comment (chuandudx) [inline firmware/lib/Gantry/Axis.cpp:228]: Do we mean 10ms or 50ms in this case? Is this still a debounce functionality?

File: firmware/lib/Gantry/Axis.cpp [hunk 4]
```diff
@@ -204,18 +226,22 @@ void isr_limit_switch_y()
...
-        NVIC_DisableIRQ(axis_y.irq_velocity);
-        NVIC_DisableIRQ(axis_y.irq_accel);
+        stop_axis(&axis_x);
         DEBUG_PRINT_VAL("Y Home Limit switch has been hit", 1);
```
Comment (chuandudx) [inline firmware/lib/Gantry/Axis.cpp:231]: nitpicky - preference for the word "reached" over "hit"

Comment (chuandudx) [inline firmware/lib/Gantry/Axis.cpp:231]: ` DEBUG_PRINT_VAL("Y axis has been homed - SUCCESS", 1);` have consistency in debug statement here?  Maybe `DEBUG_PRINT_VAL("Y Home Limit switch has been reached - SUCCESS", 1);`?

File: firmware/src/mPMTTestStand.cpp [hunk 5]
```diff
@@ -17,11 +17,13 @@ mPMTTestStand::mPMTTestStand(const mPMTTestStandIO& io) :
     thermistor_optical(io.pin_therm_optical)
 {
     // Nothing else to do
```
Comment (chuandudx) [inline firmware/src/mPMTTestStand.cpp:19]: is this just to initialize each thermistor?

File: shared/shared_defs.h [hunk 6]
```diff
@@ -28,8 +28,7 @@ typedef enum {
     STATUS_IDLE,
     STATUS_HOMING,
     STATUS_MOVING,
-    STATUS_MEASURING,
-    STATUS_LIMIT_REACHED
+    STATUS_FAULT
```
Comment (chuandudx) [inline shared/shared_defs.h:31]: Do we no longer need to indicate status for data acquisition (ie. `STATUS_MEASURING`)?

File: firmware/lib/Gantry/Axis.cpp (hunk 3 above)
Comment (noaht11) [reply firmware/lib/Gantry/Axis.cpp:228]: @oplegion ?

File: firmware/lib/Gantry/Axis.cpp (hunk 4 above)
Comment (noaht11) [reply firmware/lib/Gantry/Axis.cpp:231]: I'm gonna go through and refactor all of the motor code print statements, so not gonna bother fixing this here

Comment (noaht11) [reply firmware/lib/Gantry/Axis.cpp:231]: I'm gonna go through and refactor all of the motor code print statements, so not gonna bother fixing this here

File: shared/shared_defs.h (hunk 6 above)
Comment (noaht11) [reply shared/shared_defs.h:31]: At least for now, the Arduino has no concept of data acquisition so I removed it

File: firmware/src/mPMTTestStand.cpp (hunk 5 above)
Comment (noaht11) [reply firmware/src/mPMTTestStand.cpp:19]: Yes, like to tell each thermistor it's pin number. Is that what you're asking about?

File: firmware/lib/Gantry/Axis.cpp (hunk 1 above)
Comment (noaht11) [reply firmware/lib/Gantry/Axis.cpp:142]: No clue, but gonna go through and delete all of these later

File: firmware/lib/Gantry/Axis.cpp (hunk 3 above)
Comment (noaht11) [reply firmware/lib/Gantry/Axis.cpp:228]: Will address in future PR

**Pull request merged**